tests/Phinx/Db/Adapter/DataDomainTest.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+namespace Test\Phinx\Db\Adapter;
+
+use Phinx\Db\Adapter\AbstractAdapter;
+use Phinx\Db\Adapter\AdapterFactory;
+use Phinx\Db\Adapter\MysqlAdapter;
+use Phinx\Db\Adapter\PostgresAdapter;
+use PHPUnit\Framework\TestCase;
+
+class DataDomainTest extends TestCase
+{
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage You must specify a type for data domain type "phone_number".
+     */
+    public function testThrowsIfNoTypeSpecified()
+    {
+        $data_domain = [
+            "phone_number" => [
+                "length" => 19
+            ]
+        ];
+
+        $adapter = new MysqlAdapter(['data_domain' => $data_domain]);
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     * @expectedExceptionMessage An invalid column type "str" was specified for data domain type "phone_number".
+     */
+    public function testThrowsIfInvalidBaseType()
+    {
+        $data_domain = [
+            'phone_number' => [
+                'type' => 'str', // _Must be_ an invalid Phinx type
+                'length' => 19
+            ]
+        ];
+
+        $adapter = new MysqlAdapter(['data_domain' => $data_domain]);
+    }
+
+    /**
+     *
+     */
+    public function testConvertsToInternalType()
+    {
+        $data_domain = [
+            'phone_number' => [
+                'type' => 'string',
+                'length' => 19
+            ]
+        ];
+
+        $mysql_adapter = new MysqlAdapter(['data_domain' => $data_domain]);
+        $dd = $mysql_adapter->getDataDomain();
+
+        $this->assertEquals($data_domain['phone_number']['type'], $dd['phone_number']['type']);
+    }
+
+    /**
+     *
+     */
+    public function testReplacesLengthForLimit()
+    {
+        $data_domain = [
+            'phone_number' => [
+                'type' => 'string',
+                'length' => 19
+            ]
+        ];
+
+        $mysql_adapter = new MysqlAdapter(['data_domain' => $data_domain]);
+        $dd = $mysql_adapter->getDataDomain();
+
+        $this->assertInternalType('array', $dd['phone_number']['options']);
+        $this->assertEquals(19, $dd['phone_number']['options']['limit']);
+    }
+
+    /**
+     *
```

Comment:
`@return void` missing

Reply:
This is a test case. I've seen no test cases with type hinting or doc comments. (apart from phpunit `@expectedException`, `@covers`, etc). But I can add a `@return void` if you like. ;-)